include/sys/errno.h

```diff
@@ -176,8 +176,13 @@
 /* pseudo-errors returned inside kernel to modify return to process */
 #define EJUSTRETURN 256  /* don't modify regs, just return to userspace with */
                          /* current uctx (used by sigreturn and execve) */
-#define ERESTART 257     /* restart syscall */
 #define EPASSTHROUGH 258 /* ioctl not handled by this layer */
+/* These errors indicate when the current syscall should be restarted after
+ * being interrupted by a signal. Their semantics are the same as in Linux. */
+#define ERESTARTSYS                                                            \
+  259                      /* Restart iff no signal was caught                 \
+                            * or caught signal has SA_RESTART set */
+#define ERESTARTNOHAND 260 /* Restart iff no signal was caught */
```

Comment:
Please correct me if I am wrong but on _BSDs_ and on _Linux_ `ERESTART` exists end it is exposed to user. On the other hand `ERESTARTSYS` and `ERESTARTNOHAND` are not visible for user.
In `errno.h` we just include `sys/errno.h` so that errors are visible for user.
Could you comment that?

Reply:
In FreeBSD, `ERESTART` is internal to the kernel, as can be seen here: http://bxr.su/FreeBSD/sys/sys/errno.h#192. On Linux, `ERESTART` does seem to be exposed to userspace, but it is used very rarely.

In our case, the `ERESTART*` codes are not exposed to userspace, since they are guarded by `#ifdef _KERNEL`.

Reply:
Ok. Now I see.